rust/ql/lib/codeql/rust/controlflow/internal/ControlFlowGraphImpl.qll

```diff
@@ -391,8 +403,40 @@ class ForExprTree extends LoopingExprTree instanceof ForExpr {
   }
 }
 
-// TODO: replace with expanded macro once the extractor supports it
-class MacroExprTree extends LeafTree, MacroExpr { }
+class MacroCallTree extends ControlFlowTree instanceof MacroCall {
```

Comment:
Can't this simply be
```ql
class MacroCallTree extends StandardPostOrderTree, MacroCall {
  override AstNode getChildNode(int i) { i = 0 and result = this.getExpanded() }
}
```
?

Reply:
MacroCalls can also be statements and patterns which are pre-order. By dropping the `MacroCall` node altogether it "inherits" the order of the expanded tree.